modules/loader/src/main/java/com/opengamma/strata/loader/csv/FullSwapTradeCsvLoader.java

```diff
@@ -147,27 +152,106 @@
    * @return the parsed trade
    */
   static SwapTrade parse(CsvRow row, TradeInfo info) {
-    List<RateCalculationSwapLeg> legs = new ArrayList<>();
     // parse any number of legs by looking for 'Leg n Pay Receive'
-    Optional<String> payReceive = Optional.of(getValue(row, "Leg 1 ", DIRECTION_FIELD));
+    // this finds the index for each leg, using null for fixed legs
+    List<FloatingRateIndex> indices = new ArrayList<>();
+    Set<DayCount> dayCounts = new LinkedHashSet<>();
+    boolean missingDayCount = false;
+    String legPrefix = "Leg 1 ";
+    Optional<String> payReceiveOpt = Optional.of(getValue(row, legPrefix, DIRECTION_FIELD));
     int i = 1;
-    while (payReceive.isPresent()) {
-      legs.add(parseLeg(row, "Leg " + i + " "));
+    while (payReceiveOpt.isPresent()) {
+      // parse this leg
+      FloatingRateIndex index = parseIndex(row, legPrefix);
+      if (index != null) {
+        // floating leg
+        indices.add(index);
```

Comment:
indices.add(index) can be outside the if loop rather than here; will still add null for the fixed rate